compiler/optimizer/DeadTreesElimination.cpp

```diff
@@ -883,6 +901,45 @@ int32_t TR::DeadTreesElimination::process(TR::TreeTop *startTree, TR::TreeTop *e
          }
       }
 
+   for (auto it = anchors.begin(); it != anchors.end(); ++it)
+      {
+      TR::Node *anchor = it->tree->getNode();
+      TR::Node *load = anchor->getChild(0);
+      if (load->getReferenceCount() > 1)
+         continue;
+
+      TR_ASSERT_FATAL(
```

Comment:
While it's likely that the child of a compressed refs anchor is an indirect load, is there a reason why the check at the point where the compressed refs anchor is added to the "anchors" data structure is only isLoad ? i.e. is there a possibility that this assert fails in some obscure case when the indirect load has been changed to a direct load and the compressed refs anchor has not been deleted ? Why not check for isLoadIndirect at the point of the addition to "anchors" ?

Reply:
Tree verification passes do not run in production builds, while the above assert will fire. I do see your point however, and I agree.

Reply:
Updated to remove the assertion and instead look specifically for `isLoadIndirect()` when queueing up the trees